src/diffusers/schedulers/scheduling_ald.py

```diff
@@ -0,0 +1,194 @@
+        self.final_noise_sigma = None
+        self.step_lr = step_lr
+
+        # setable values
+        self.num_inference_steps = None
+        self.timesteps = torch.from_numpy(np.arange(0, num_train_timesteps)[::-1].copy())
+
+        self.set_sigmas(num_train_timesteps, sigma_min, sigma_max)
+
+    def scale_model_input(self, sample: torch.FloatTensor, timestep: Optional[int] = None) -> torch.FloatTensor:
+        """
+        Ensures interchangeability with schedulers that need to scale the denoising model input depending on the
+        current timestep.
+
+        Args:
+            sample (`torch.FloatTensor`): input sample
+            timestep (`int`, optional): current timestep
+
+        Returns:
+            `torch.FloatTensor`: scaled input sample
+        """
+        return sample
+
+    def set_timesteps(self, num_inference_steps: int, device: Union[str, torch.device] = None):
+        """
+        Sets the continuous timesteps used for the diffusion chain. Supporting function to be run before inference.
+
+        Args:
+            num_inference_steps (`int`):
+                the number of diffusion steps used when generating samples with a pre-trained model.
+
+        """
+        num_inference_steps = min(self.config.num_train_timesteps, num_inference_steps)
+        self.num_inference_steps = num_inference_steps
+        timesteps = np.arange(
+            0, self.config.num_train_timesteps, self.config.num_train_timesteps // self.num_inference_steps
+        )[::-1].copy()
+        self.timesteps = torch.from_numpy(timesteps).to(device)
+
+    def set_sigmas(self, num_inference_steps: int, sigma_min: float = None, sigma_max: float = None):
+        """
+        Sets the noise scales used for the diffusion chain. Supporting function to be run before inference.
+
+        The sigmas control the weight of the `drift` and `diffusion` components of sample update.
+
+        Args:
+            num_inference_steps (`int`):
+                the number of diffusion steps used when generating samples with a pre-trained model.
+            sigma_min (`float`, optional):
+                initial noise scale value (overrides value given at Scheduler instantiation).
+            sigma_max (`float`, optional): final noise scale value (overrides value given at Scheduler instantiation).
+
+        """
+        sigma_min = sigma_min if sigma_min is not None else self.config.sigma_min
+        sigma_max = sigma_max if sigma_max is not None else self.config.sigma_max
+
+        if self.timesteps is None:
+            self.set_timesteps(num_inference_steps)
+
+        self.sigmas = torch.tensor(
+            torch.exp(torch.linspace(torch.log(sigma_min), torch.log(sigma_max), num_inference_steps)),
+            dtype=torch.float32,
+        )
+
+        self.final_noise_sigma = self.sigmas[-1]
+
+    def step(
+        self,
+        model_output: torch.FloatTensor,
+        timestep: int,
```

Comment:
@anton-l can you check here if this fits with our scheduler API in general ? It seems like `t` is more used as an index then the corresponding timestep no?

Reply:
Gently ping here @anton-l

Reply:
@natolambert I think we have to change this here - We don't really want to pass timesteps as indexes for sigma except for the indexes and sigma are always identical or this use case is very special. 

Overall, if you don't think we can also adapt the scheduler later on - fine for me as well

Reply:
Did a literature review to track this down. So what is happening here is that [Score matching with Langevin dynamics](https://arxiv.org/abs/1907.05600) (SMLD) (this paper) is an intellectual predecessor of [SDE VE/P](https://arxiv.org/abs/2011.13456), which also uses this `self.sigma[timestep]` API. 

Does this change any of your opinions @patrickvonplaten @anton-l. I'm not sure the right way to do that. These two schedulers are the only that do that (unless SDE VP gets finished)